scipy/stats/tests/test_contingency.py

```diff
@@ -196,3 +196,62 @@ def test_chi2_contingency_bad_args():
     obs = np.empty((0, 8))
     assert_raises(ValueError, chi2_contingency, obs)
 
+
+def test_bad_association_args():
+    # Invalid Chi Squared Stat Value Type
+    assert_raises(ValueError, association, [[1, 2], [3, 4]], "x", "tschuprow")
+    # Invalid Test Statistic
+    assert_raises(ValueError, association, [[1, 2], [3, 4]], 2.55, "X")
+    # Invalid array shape
+    assert_raises(ValueError, association, [[[1, 2]], [[3, 4]]], 2.55, "cramer")
+
+
+def test_cramersv():
```

Comment:
Is a 2-D test with shape (3, 5) all that need needs testing? What about n-D arrays, arrays with too few observations (or empty ones); other unexpected input (e.g. float dtypes?)

Reply:
These statistics are generally only defined for 2-d arrays, so testing with 2-d input only should be fine.

More type validation is certainly possible.  If the calculation of the chi2 statistic is incorporated into the function, `chi2_contingency` will raise an error if any value in the contingency table is negative.  But it won't complain about floating point values (even values that are not integers are accepted).  I'd be OK with tightening that up and accepting only integer data types in this new function. (I don't know if such a change to `chi2_contingency` itself would cause problems out in the wild.)

Reply:
Made these changes I have it raising a value error if it cant cast the array as int. I didn't know how to handle cases like 1.5 which would become 1 in this case. I guess I can be stricter with my type check and go by individual value it just seemed excessive and likely more appropriate upstream.